source/MaterialXCore/Traversal.h

```diff
@@ -14,8 +14,15 @@
 namespace MaterialX
 {
 
+class Element;
+class Material;
+
+/// A shared pointer to an Element
```

Comment:
Although I see the consistency argument in providing Doxygen comments for all type aliases, these are really forward declarations of the "official" type aliases for ElementPtr and MaterialPtr in Element.h and Material.h.  From that perspective, it may be better to omit these duplicate Doxygen comments, leaving the versions in Element.h and Material.h as authoritative.

Reply:
I see. I didn't realize they were the only duplicates, excepting the class-specific aliases. I dropped the commit adding the duplicate doxygen from the PR.